scripts/myMangaDex.js

```diff
@@ -1654,6 +1656,15 @@ class MyMangaDex {
 				}
 				return chap != chapter;
 			});
+			let prev = Math.floor(last) - 1;
+			// always choose one that is within 1 range
+			// (unavoidable) error: either use floored, leads to false next chapter when using sub-chapters
+			//											or add 0.99999, leads to correct next chapter but no current chapter
+			if (this.manga.last == -1 || Math.floor(this.manga.last) < prev) {
+				this.manga.last = prev + 0.99999;
+				this.manga.currentChapter.chapter = prev + 0.99999;
+				this.manga.lastMangaDexChapter = prev + 0.99999;
+			}
```

Comment:
As can be seen in the added comment, there is an unsolvable problem here. I chose this solution because it correctly marks the next chapter in any case and the current in no case, unlike the other solution which only marks next and current correctly if the manga uses whole chapters and nothing correctly in other cases.

Reply:
The problem with that is, that I have also seen chapters like .9
Also you can't round because then it's always the base chapter, so the one marked as next could be a subchapter. I think a better solution is like I did it, because it will guarantee that the chapter marked as next is higher, just maybe a little higher than intended (vs lower than intended).
Ultimately removing the `+0.99999` is the same as what you proposed.
It's your choice really. There shouldn't be that many subchapters around anyways and maybe correctly selecting the current and next chapter in most cases is more important...

Reply:
Well, I trust you with that then, should have done this 12 days ago, sorry for the delay.